packages/lit-dev-content/site/blog/2022-07-19-lit-for-polymer-users.md

```diff
@@ -0,0 +1,1013 @@
+
+In most cases, most of the migration process can be automated using the Polymer modulizer tool. For more information, see the [Polymer 3.0 upgrade guide](https://polymer-library.polymer-project.org/3.0/docs/upgrade).
+
+
+### Language support
+
+Polymer used features from the ECMAScript 2015 version of the JavaScript spec. If you started from one of the Polymer starter kits, your toolchain may not support newer JavaScript.
+
+Lit uses features from ECMAScript 2019 (and some Lit example code may include newer language features). You'll need to update your tools if they don't handle these newer JavaScript features. For example:
+
+*   Public instance fields and public static fields. These are widely used in the example code:
+
+    ```js
+    static styles = [ css`:host { display: block; }` ];
+    ```
+
+*   Async and await, which can be used to simplify promise-based code. For example:
+
+    ```js
+    async _loginClickHandler() {
+      this.loggedIn = true;
+      // Wait for `loggedIn` state to be rendered to the DOM
+      await this.updateComplete;
+      this.dispatchEvent(new Event('login'));
+    }
+    ```
+
+For more information on language requirements for Lit, see [Requirements](https://lit.dev/docs/tools/requirements/).
+
+
+### Removing two-way bindings { #removing-two-way-bindings }
+
+Polymer's two-way binding effectively tightly couples a host property with a child property. This tight coupling creates a number of issues, so the team chose not to add two-way binding to Lit.
+
+Removing two-way bindings **before** migrating to Lit will reduce the complexity of your migration, and let you test your application in Polymer without two-way bindings before starting your migration.
+
+If your application doesn't use two-way binding, you can skip this section.
+
+
+#### Issues with two-way binding
+
+For two-way bindings, Polymer uses its own protocol, which has three main components:
+
+*   A binding from host to child.
+
+*   An event listener that handles property change events from the child element.
+
+*   A facility for the child to automatically fire change events when properties change (`notify: true`).
+
+This last item is the most problematic. Components fire change events for any change to a property, and each change event is handled synchronously.  Widespread use of two-way binding across an entire application can make it hard to reason about data flow and the order in which components update themselves.
+
+Ideally an event is a discrete signal sent to communicate an explicit change that isn't otherwise easily observable. Sending an event as a side-effect of setting a property as Polymer makes the communication potentially redundant and implicit. This implicit behavior, in particular, can make data flow hard to understand.
```

Comment:
```suggestion
Ideally an event is a discrete signal sent to communicate an explicit change that isn't otherwise easily observable. Sending an event as a side-effect of setting a property as Polymer does makes the communication potentially redundant and implicit. This implicit behavior, in particular, can make data flow hard to understand.
```